src/evict/evict_lru.c

```diff
@@ -1522,6 +1522,14 @@ __evict_walk(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue)
             }
             __wt_spin_unlock(session, &cache->evict_walk_lock);
             WT_ERR(ret);
+            /*
+             * If there is a checkpoint thread gathering handles, which means it is holding the
+             * schema lock, then there is often contention on the evict walk lock with that thread.
+             * If eviction is not in aggressive mode, sleep a bit to give the checkpoint thread a
+             * chance to gather its handles.
+             */
+            if (F_ISSET(conn, WT_CONN_CKPT_GATHER) && !__wt_cache_aggressive(session))
+                __wt_sleep(0, 10);
```

Comment:
I think a statistic here would be useful - it would be nice to get signal as to why the eviction server has slowed down if that starts causing problems.